### DISCUSSION TITLE
Question about Sample Code in OPA Document

### DISCUSSION THREAD

**Discussion by Ronnie-personal** [open-policy-agent/community#343] [OPA and Rego] [answered]
I have a question regarding the sample code provided in the following link: https://www.openpolicyagent.org/docs/latest/extensions/.
Do we need to assign 'err' in the second 'if' statement, such as in the following code: 'else if err := ast.As(b.Value, &repo); err != nil { ... }'?

		func(bctx rego.BuiltinContext, a, b *ast.Term) (*ast.Term, error) {

			var org, repo string

			if err := ast.As(a.Value, &org); err != nil {
				return nil, err
			} else if ast.As(b.Value, &repo); err != nil {
				return nil, err
			}

**Comment by anderseknert** [comment, marked as answer]
Yes, that looks rather wrong to me. Does it even compile? 🤔 If you've modified the example to make it work, a PR would be welcome! :)

**Reply by Ronnie-personal** [reply]
It compiles, but just does not perform as intended.

I have updated the sample code and created PR https://github.com/open-policy-agent/opa/pull/5666

**Reply by anderseknert** [reply]
Fantastic 👍
